astropy/io/fits/util.py

```diff
@@ -727,7 +728,7 @@ def _words_group(s, width):
 
     # locations of the blanks
     blank_loc = np.nonzero(arr == b" ")[0]
-    offset = 0
+    offset = 0 if first_width is None else first_width - width
```

Comment:
Setting `offset` at the start of `_words_group()` does not actually accomplish anything, because `offset` is reset in the while loop below.

Reply:
That is as intended: further in the loop, we know the card starts with `CONTINUE`, so a reset is proper.

I should have properly tested the single long word, though. And I clearly missed the comment case...